_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,285 @@
+
+## 3. Writing unit tests using GoogleTest
+
+For this episode, we will consider the same example `fibonacci.cpp` that we used previously. There we identified 3 possible cases when using the `recursive_fibonacci` function for which we wrote some manual tests:
+
+- `n` equal to 0 or 1
+- `n` a negative number
+- `n` a positive number greater than 1
+
+To demonstrate how to use GoogleTest, we will simply be converting these tests that we wrote manually in the same file using GoogleTest framework. Section 5 below will describe the anatomy of a test, but first let's see what they look like in the practical case.
+For using GoogleTest in your code, you need to follow the following steps in general.
+
+### 3.1. Adding the required header files
+
+The first step would be to add the required header files in your program. For GoogleTest, you would need to add the following line in your code.
+
+```cpp
+#include "gtest/gtest.h"
+```
+
+### 3.2. Create your tests
+
+The next step would be to to define your test. GoogleTest uses the following convention in naming various tests.
+
+```cpp
+TEST(TestSuiteName, TestName)
+{
+    // Test logic and assertions
+}
+```
+
+The different parts in the above cell have the following meaning
+
+- `TEST`: This is the macro provided by the GoogleTest framework to define a test case. (For more details about macros, see [Macros in C++](https://gcc.gnu.org/onlinedocs/cpp/Macros.html))
+- `TestSuiteName`: This is the name of the test suite or test fixture. It groups related tests together. It should be a valid C++ identifier.
+- `TestName`: This is the name of the individual test case within the test suite. It should also be a valid C++ identifier.
+- `Test logic and assertions`: This is the body of the test case where you write the actual test code, including any necessary assertions to verify the expected behaviour. The section `Test Assertions` describes more about various assertions available in GoogleTest.
+
+### 3.3. Initialise GoogleTest framework in your main function
+
+When you install GoogleTest you would get two libraries namely `libgtest.a` and `libgtest_main.a` (The extension `.a` means it is a static library and is applicable for Linux based systems). The first one, i.e. `libgtest.a`, is the library which provides all the necessary testing features such as assertions, test discovery, collection of results etc. The second library, i.e. `libgtest_main.a`, provides a main function so that you do not need to write your own main function for testing. For more details, refer to https://stackoverflow.com/questions/6457856/whats-the-difference-between-gtest-lib-and-gtest-main-lib.  
```

Comment:
I wouldn't have a "naked" link in the text. Give it a text description.

Also, it would be better if you could link to the GoogleTest documentation, but this is also fine.

Reply:
resolved in aa1a24a83286f710250741a56e9d87b00d10969f